### PR TITLE
Event filter

### PR DESCRIPTION
**Pull request by nbuonin**

### CODE REVIEW

**Comment by nikolas** [comment]
One change in behavior here that I'm not sure was intentional:

Currently on the master branch, refreshing a page with URL params selects the appropriate options in the select box:
2017-08-31-093435_1193x600_scrot

On this branch, the search results are selected but the select boxes remain at the default value:
2017-08-31-093448_1285x663_scrot

File: utils.js [hunk 1]
```diff
@@ -297,7 +262,33 @@ CTLEventUtils.populateURLParams = function(paramsArray) {
...
+    for (var i = 0; i < paramsArray.length; i++) {
+        if (paramsArray[i].key == param) {
+            value = paramsArray[i].value;
+            // if the param is supposed to be a date, instantiate a Date object
+            if (param == 'start' || param == 'end') {
+                value = new Date(value);
+            }
+        }
+    }
+    return value;
```
Comment (nikolas) [inline utils.js]: I would also add `return value;` to after line 284, since you don't need to iterate through the rest of the array once the value is found.

**Comment by nbuonin** [comment]
Yep, the populating of the dropdowns is a regression I'll fix.

File: utils.js (hunk 1 above)
Comment (nbuonin) [reply utils.js]: Got it, fixed.

File: main.js [hunk 2]
```diff
@@ -99,116 +77,92 @@
...
 
         refreshEvents(filteredEvents, 1);
     };
 
+    var filterEventHandler = function() {
+        // Clear the events
+        $('#calendarList').empty();
+        // Then get the vars ready
+        var $searchWrapper = $('#search-wrapper');
+        var q = null;
```
Comment (sdreher) [inline main.js]: These variable initializations are redundant: q, loc, audience, startDate, endDate.

File: utils.js [hunk 3]
```diff
@@ -20,12 +30,17 @@ CTLEventUtils.formatShortDate = function(d) {
  * @return = an array of events that match the query
  */
 CTLEventUtils.searchEvents = function(allEvents, index, q) {
-    var results = index.search(q);
+    if (q == null || q == '') {
```
Comment (sdreher) [inline utils.js]: `if (q)` should serve the same purpose here. See [Logical Operators](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Logical_Operators)

also - question on Ln. 70, `if (audience === null || audience === 'null') `. Could audience evaluate to 'null'?

File: main.js (hunk 2 above)
Comment (nbuonin) [reply main.js]: yes, they are, I've removed them

File: utils.js (hunk 3 above)
Comment (nbuonin) [reply utils.js]: I don't think it could, but I've kept it around to standardize the filter behavior - if there are no matches it returns everything. As I'm writing this out though, this seems like an unexpected behavior.  Should I remove them?

Comment (nikolas) [reply utils.js]: A rule of thumb I follow is to always use triple-equal in javascript boolean statements. `(q == null || q == '')` is redundant because, since the empty string is `==` to null, the second clause here doesn't do anything.

If you are just trying to find out if the value is "falsey", `if (q)` should suffice, like Susan points out. Otherwise, explicitly using `===` will make your code clearer.

Comment (sdreher) [reply utils.js]: on the audience question - I think the shortcut of `if (null) { return all }` is fine. I was more curious about the test for a string with `'null'`. Is that a possible value?

Comment (nikolas) [reply utils.js]: Oh, that's actually incorrect. In my browser, `'' == null` is actually false. Just when you think you understand it, javascript can do something unexpected.

But anyways, always using `===` is still a good rule of thumb.

Comment (nbuonin) [reply utils.js]: @sdreher No, its not, but it was prior to refactoring. I was doing something silly like this: https://github.com/ccnmtl/calendar-widget/pull/80/files#diff-a8a494a0e94448e7e4cce52d0cb1ce9eL42
